libraries/rush-lib/src/utilities/objectUtilities.ts

```diff
@@ -114,3 +114,56 @@ function cloneDeepInner<TObject>(obj: TObject, seenObjects: Set<unknown>): TObje
     return obj;
   }
 }
+
+/**
+ * Performs a partial deep comparison between `obj` and `source` to
+ * determine if `obj` contains equivalent property values.
+ */
+export function isMatch<TObject>(obj: TObject, source: TObject): boolean {
```

Comment:
Do we need this?  It seems to be only used by this method of `BaseFlag`:

```ts
  /**
   * Merge new data into current state by "merge"
   */
  public mergeFromObject(data: JsonObject): void {
    if (isMatch(this._state, data)) {
      return;
    }
    merge(this._state, data);
    this._isModified = true;
  }
```

However I don't see anything calling `mergeFromObject()`, and the use case seems unclear.  In general the `last-install.flag` file may have been written by a Rush release that is much older or much newer than the current Rush, which may write different fields to that file. Can we safely merge them?